pkg/management/component.go

```diff
@@ -495,6 +498,49 @@ func (c *component) GetGroupsOfUser(ctx context.Context, realmName, userID strin
 	return groupsRep, nil
 }
 
+func (c *component) SetTrustIDGroups(ctx context.Context, realmName, userID string, groupNames []string) error {
+	var accessToken = ctx.Value(cs.CtContextAccessToken).(string)
+
+	// validate the input - trustID groups must be valid
+	var extGroupNames []string
+	for _, groupName := range groupNames {
+		var match = false
+		for _, authzedGrpName := range c.authorizedTrustIDGroups {
```

Comment:
You can avoid this whole `for` statement by using a map as previously explained

Reply:
done